sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaSensor.java

```diff
@@ -0,0 +1,269 @@
+                        .toString();
+                String fileContent = getFileContent(analyzedResultsFilePath);
+                reportFileContent(context, pathAndInputFiles, fileContent);
+            } else {
+                logger.info("Unable to analyze ballerina file batch!");
+            }
+        } catch (IOException | InterruptedException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private String getFileContent(String analyzedResultsFilePath) {
+        String fileContent = "";
+        try (
+                FileReader fileReader = new FileReader(analyzedResultsFilePath, StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)
+        ) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+            fileContent = stringBuilder.toString();
+        } catch (IOException ex) {
+            logger.info("Unable to retrieve analysis results!");
+        }
+        return fileContent;
+    }
+
+    private void reportFileContent(SensorContext context,
+                                   Map<String, InputFile> pathAndInputFiles,
+                                   String fileContent) {
+        JsonArray balScanOutput = null;
+        try {
+            balScanOutput = JsonParser.parseString(fileContent).getAsJsonArray();
+        } catch (Exception ignored) {
+            logger.info("Unable to report analysis results!");
+        }
+
+        boolean reportingSuccessful = reportAnalysisIssues(context, balScanOutput, pathAndInputFiles);
+        if (reportingSuccessful) {
+            logger.info("Ballerina analysis successful!");
+        } else {
+            logger.info("Unable to analyze ballerina file batch!");
+        }
+    }
+
+    public boolean reportAnalysisIssues(SensorContext context,
+                                        JsonArray analysisIssues,
+                                        Map<String, InputFile> pathAndInputFiles) {
+        boolean reportingSuccessful = false;
+        if (analysisIssues != null) {
+            for (JsonElement issueElement : analysisIssues) {
+                JsonObject issue = issueElement.getAsJsonObject();
+                String absoluteFilePath = issue.get(ISSUE_FILE_PATH).getAsString();
+                InputFile inputFile = pathAndInputFiles.get(absoluteFilePath);
+                String issueType = issue.get(ISSUE_SOURCE).getAsString();
+                switch (issueType) {
+                    case BUILT_IN -> reportIssue(inputFile, context, issue);
+                    case EXTERNAL -> reportExternalIssue(inputFile, context, issue);
+                    default -> logger.info("Invalid Issue Format!");
+                }
+            }
+            reportingSuccessful = true;
+        }
+        return reportingSuccessful;
```

Comment:
If analysisIssues is null return false, then move the rest of the logic out. The `reportingSuccessful` variable is unnecessary.